### DISCUSSION TITLE
How to encrypt using passwords starting with "-"

### DISCUSSION THREAD

**Discussion by sz5000** [qpdf/qpdf#873] [Q&A]
If password starts with -, qpdf outputs the following error:

`unrecognized argument -test (encryption options must be terminated with --)`

qpdf is called with @- to read args from stdin.

```
--encrypt
XXXX
YYYY
256
[options]
--
```

The question now is how do we use passwords that start with "-"?
Can we quote or otherwise encode them so that qpdf does not treat them as switch?

If either XXXX or YYYY is something like "-test", then the command line error mentioned above is shown. 

```
--encrypt
-test
YYYY
256
[options]
--
```

using `--password-mode=hex-bytes` seems to be problematic because of encoding based on used encryption mode.

**Comment by jberkenbilt** [comment]
I think this is a bug, but there is a workaround. `--password-mode` applies to the password used to open the input file, not the password used for encryption. Right now, the only workaround I can think of is to use job json. For example, if you create this JSON file called enc.json:
```json
{
  "encrypt": {
    "userPassword": "--test",
    "ownerPassword": "owner",
    "256bit": {}
  }
}
```
then you can run `qpdf --job-json-file=enc.json in.pdf out.pdf`

Anyway, it is a bug that qpdf is treating the thing that appears in that position as an argument. I will create an issue from this. Thanks.

**Comment by jberkenbilt** [comment]
A fix is coming in qpdf 11.7.0. See #874 for details.
